src/paymasterclient.ts

```diff
@@ -49,19 +53,86 @@ export type Bundle = {
 }
 
 export class PaymasterClient extends ethers.JsonRpcProvider {
-  constructor(url?: string | FetchRequest, network?: Networkish, options?: JsonRpcApiProviderOptions) {
+  private privatePolicyUUID?: string
+
+  private constructor(
+    url?: string | FetchRequest,
+    network?: Networkish,
+    options?: JsonRpcApiProviderOptions,
+    privatePolicyUUID?: string
+  ) {
     super(url, network, options)
+    this.privatePolicyUUID = privatePolicyUUID
+  }
+
+  // Static method to create a new standard PaymasterClient
+  static new(
+    url?: string | FetchRequest,
+    network?: Networkish,
+    options?: JsonRpcApiProviderOptions
+  ): PaymasterClient {
+    return new PaymasterClient(url, network, options)
+  }
+
+  // Static method to create a new PaymasterClient with private policy
+  static newPrivatePaymaster(
+    url: string | FetchRequest,
+    privatePolicyUUID: string,
+    network?: Networkish,
+    options?: JsonRpcApiProviderOptions
+  ): PaymasterClient {
+    return new PaymasterClient(url, network, options, privatePolicyUUID)
   }
 
   async chainID(): Promise<string> {
     return await this.send('eth_chainId', [])
   }
 
   async isSponsorable(tx: TransactionRequest): Promise<IsSponsorableResponse> {
+    const policyUUID = this.privatePolicyUUID
+    if (policyUUID) {
+      const newConnection = this._getConnection()
+      newConnection.setHeader("X-MegaFuel-Policy-Uuid", policyUUID)
+      const sponsorProviderWithHeader = new ethers.JsonRpcProvider(
```

Comment:
`sponsorProviderWithHeader` the name `sponsor` is not proper here. 

if no new header to set here, can we not do `new ethers.JsonRpcProvider`, but reuse the current one.

Reply:
sponsorProviderWithHeader the name sponsor is not proper here. ------> rename to provider 

if no new header to set here, can we not do new ethers.JsonRpcProvider, but reuse the current one.
------------>
<img width="588" alt="image" src="https://github.com/user-attachments/assets/4793194f-3740-4cc5-b02c-b8b130a42859">

`if (policyUUID)`
if `policyUUID` is undefined, null, or an empty string. In any of these cases, the program will skip the if block and directly execute:  `return await this.send('pm_isSponsorable', [tx])`
This way, we didn't create a `new ethers.JsonRpcProvider` in cases where we don't need to set a header.